lib/checkclass.cpp

```diff
@@ -1655,11 +1655,14 @@ void CheckClass::virtualDestructor()
         if (scope->definedType->derivedFrom.empty()) {
             if (printInconclusive) {
                 const Function *destructor = scope->getDestructor();
-                if (destructor && !destructor->hasVirtualSpecifier()) {
-                    for (const Function &func : scope->functionList) {
-                        if (func.hasVirtualSpecifier()) {
-                            inconclusiveErrors.push_back(destructor);
-                            break;
+                if (destructor) {
+                    if(!((destructor->hasVirtualSpecifier() && destructor->access == AccessControl::Public) ||
+                        (!destructor->hasVirtualSpecifier() && destructor->access == AccessControl::Protected))) {
```

Comment:
hmm.. is it *bad* for some reason with a virtual protected destructor?

Reply:
I went for a strict application of the guideline. They discuss the case of the virtual protected destructor: in it self it is not _bad_, but the use case is quite unique:
> We can imagine one case where you could want a protected virtual destructor: When an object of a derived type (and only of such a type) should be allowed to destroy another object (not itself) through a pointer to base. We haven't seen such a case in practice, though.

Since the message is already _inconclusive_ and this is an uncommon use case I chose to go with the stricter version to warn the user that he is doing an unusual thing.
Maybe it would better suit a lower level message in this case (_style_?).

If you want to drop the "not virtual" for the protected descructor, I have no issue with that).

Reply:
> If you want to drop the "not virtual" for the protected descructor, I have no issue with that.

thanks. then please adjust it so we are less strict about this.

I do agree it would be an unusual use case. ;-)